test/unix/force_merge.ml

```diff
@@ -12,20 +12,58 @@ let test_find_present t =
       match Index.find t k with
       | res ->
           if not (res = v) then
-            Alcotest.fail (Printf.sprintf "Replacing existing value failed.")
+            Alcotest.fail "Replacing existing value failed."
       | exception Not_found ->
-          Alcotest.fail
-            (Printf.sprintf "Inserted value is not present anymore: %s." k))
+          Alcotest.failf "Inserted value is not present anymore: %s." k)
     tbl
 
 let test_one_entry r k v =
   match Index.find r k with
   | res ->
-      if not (res = v) then
-        Alcotest.fail (Printf.sprintf "Replacing existing value failed.")
+      if not (res = v) then Alcotest.fail "Replacing existing value failed."
   | exception Not_found ->
-      Alcotest.fail
-        (Printf.sprintf "Inserted value is not present anymore: %s." k)
+      Alcotest.failf "Inserted value is not present anymore: %s." k
+
+let test_fd () =
+  let pid = string_of_int (Unix.getpid ()) in
+  let fd_file = "tmp" in
+  let lsof_command = "lsof -a -s -p " ^ pid ^ " > " ^ fd_file in
```

Comment:
Not all Unix systems have `lsof`; mine didn't. The tests probably shouldn't be failing on such systems; would be better to just skip the test altogether.